Calendar Export Functionality

tenacke opened the pull request:
Added calendar export functionality that allows students to download their course schedules as ICS files for import into calendar applications.

### Features
- **Calendar Export Button**: Located in CourseList component
- **ICS File Generation**: Creates standard calendar files with course details and locations
- **Holiday Support**: Automatically excludes classes on holidays (full-day and time-based)
- **Future Semester Only**: Button disabled for past semesters

### Maintenance
- **Add New Semester**: Update semester-dates.json accordingly
- **Add Holidays to Exclude Calendar**: Update holidays.json

kilicbaran commented on src/lib/CalendarExport.svelte:
```diff
@@ -0,0 +1,384 @@
+  onclick={downloadCalendar}
+  disabled={!canExportCalendar}
+  title={canExportCalendar 
+    ? "Open schedule in your calendar app" 
```
I found this message confusing. Does clicking this button open the file in a calendar app? Or did you mean it like after clicking this button do this step. I think it would be better if button tooltip explains the action it actually does.

kilicbaran replied:
`Export Calendar File (.ics)` or `Download Calendar File (.ics)` is okey with me

kilicbaran commented on src/lib/CalendarExport.svelte:
```diff
@@ -0,0 +1,384 @@
+    
+    // Show user instruction for manual import if needed
+    setTimeout(() => {
+      if (confirm("Calendar file downloaded! Would you like instructions on how to import it into your calendar app?")) {
```
nitpick: Do we have to prompt the user twice? Is there anything that make this necessary? Wouldn't it be simpler to gve all the description at once. What do you think?

tenacke replied:
You are right. When I also review it now, I can tell that sending pop-up is entirely unnecessary. I propose putting a clickable text saying 'Instructions to add'. Clicking that text opens the instructions. I can implement this way if you also agree with me.

kilicbaran replied:
I think a single alert with both confirm and alert description concatenated would suffice. In your proposal, you again made it two steps (alert + instruction click). I'm just trying to understand. Is there any need for this? Do you think it will look long or anything else?

tenacke replied:
In either way, Confirm + instructions click is unnecessary. But now I think confirm alert is also unnecessary. That's what I suggest. I also want to remove the confirmation alert. The user can easily know successful states by seeing some file downloading. Instruction click is what I wanted to add to keep the page as simple as possible. That's why I insist on hiding into a click.

kilicbaran replied:
Okey, I didn't get where you will place it but you can do it. I will get it when I see it I guess. You probably it somewhere around the button?

tenacke commented:
> I wish there was an easy way to handle them.

I want to mark it as a later work for now. I didn't want to implement both automatic and manual holiday configurations. There should be open source calendar feeds to fetch the current year's holidays. I can work on them later but still they are not reliable as the academic calendar of Boun since there are also semester breaks which are not a national holiday.

tenacke commented:
I guess I added the new changes. Here is the outline of the latest commit:

- Combined holidays.json into semester-dates.json for single source of data
- Uses weekly recurring events with holiday exclusions instead of individual events
- Modified UI with "How to import?" clickable text
